### PR TITLE
Use ODS for StableHLO types

### LINKED ISSUE #1876: Generate StableHLO TokenType with ODS
Context: https://github.com/openxla/stablehlo/pull/1871#issuecomment-1846259583

### PR DESCRIPTION
**Pull request by mlevesquedion**
Closes #1876.

### CODE REVIEW

File: stablehlo/dialect/StablehloOps.cpp [hunk 1]
```diff
@@ -76,6 +76,9 @@ limitations under the License.
 #include "stablehlo/dialect/StablehloOps.h.inc"
 #include "stablehlo/dialect/TypeInference.h"
 
+#define GET_TYPEDEF_CLASSES
+#include "stablehlo/dialect/StablehloTypeDefs.cpp.inc"
```
Comment (GleasonK) [inline stablehlo/dialect/StablehloOps.cpp:80]: Can we put this right next to Enum / Attr includes?

File: stablehlo/dialect/CMakeLists.txt [hunk 2]
```diff
@@ -135,6 +135,8 @@ mlir_tablegen(StablehloEnums.h.inc -gen-enum-decls)
 mlir_tablegen(StablehloEnums.cpp.inc -gen-enum-defs)
 mlir_tablegen(StablehloAttrs.h.inc -gen-attrdef-decls)
 mlir_tablegen(StablehloAttrs.cpp.inc -gen-attrdef-defs)
+mlir_tablegen(StablehloTypeDefs.h.inc -gen-typedef-decls --typedefs-dialect=stablehlo)
```
Comment (GleasonK) [inline stablehlo/dialect/CMakeLists.txt:138]: Looks like we're missing bazel build rules?

**Review by GleasonK — CHANGES_REQUESTED** [review]

File: stablehlo/dialect/StablehloOps.cpp (hunk 1 above)
Comment (GleasonK) [reply stablehlo/dialect/StablehloOps.cpp:80]: Ah is clang format moving this up? Not sure why `#include "stablehlo/dialect/StablehloOps.h.inc"` got moved up into the main include line, if we can move this down as well after all other headers

Comment (mlevesquedion) [reply stablehlo/dialect/StablehloOps.cpp:81]: Done.

No I put it there myself, sorry about the bad placement. I moved the Ops include down as well, PTAL.

File: stablehlo/dialect/CMakeLists.txt (hunk 2 above)
Comment (mlevesquedion) [reply stablehlo/dialect/CMakeLists.txt:138]: 🤦 Ah yep, thank you! Done. Sorry, I was only running the CMake build script, not the bazel build script.

**Review by GleasonK — APPROVED** [review]

**Pull request merged**